src/utilities/buildASTSchema.js

```diff
@@ -164,57 +165,31 @@ export function buildASTSchema(ast: DocumentNode): GraphQLSchema {
     }
   }
 
-  let queryTypeName;
-  let mutationTypeName;
-  let subscriptionTypeName;
+  let operationTypes = {
```

Comment:
nit: Should be `const`

Reply:
@asiandrummer It's intentional since on line 174:
```js
operationTypes = {};
```

This is because schema definition is optional but if it present you should use it instead on relaying on type names.

Reply:
```
const operationTypes  = schemaDef ? { ... } : {};
```
Above seems better to me, but it's more of a personal pref to me - again it's a nit

Reply:
@asiandrummer Great idea 👍 
I refactor out `getOperationTypes` function so it become:
```js
const operationTypes  = schemaDef ? getOperationTypes(schemaDef) : {
   ... 
};
```